### PR TITLE
Aliu/erisc launch msg

### PR DESCRIPTION
**Pull request by aliuTT**

### CODE REVIEW

File: tt_metal/hw/inc/wormhole/eth_l1_address_map.h [hunk 1]
```diff
@@ -30,11 +30,14 @@ struct address_map {
   static constexpr std::int32_t COMMAND_Q_BASE = L1_EPOCH_Q_BASE + FIRMWARE_SIZE;
   static constexpr std::int32_t DATA_BUFFER_BASE = COMMAND_Q_BASE + COMMAND_Q_SIZE;
   static constexpr std::int32_t TILE_HEADER_BUFFER_BASE = DATA_BUFFER_BASE + DATA_BUFFER_SIZE;
-  static constexpr std::int32_t PRINT_BUFFER_ER = COMMAND_Q_BASE - 224;
-  static constexpr std::int32_t ERISC_MEM_MAILBOX_BASE = PRINT_BUFFER_ER - 128;
+  static constexpr std::int32_t ERISC_MEM_MAILBOX_BASE = 0x10e94;  // See dev_msgs.h and dev_mem_map.h for restrictions
```
Comment (tt-dma) [inline tt_metal/hw/inc/wormhole/eth_l1_address_map.h:33]: Is there a way we could write this in a slightly less opaque way? Like `COMMAND_Q_BASE - <Some hard-coded value>` with a comment saying how we need to align launch message.

**Review by tt-dma — APPROVED** [review]

File: tt_metal/hw/inc/wormhole/eth_l1_address_map.h (hunk 1 above)
Comment (aliuTT) [reply tt_metal/hw/inc/wormhole/eth_l1_address_map.h:33]: Yea I thought about it.. I think this is same logic as tensix, where it's written like this:
```
MEM_MAILBOX_BASE = 20;
yadayada

ASSERT(mailbox->launch is aligned)
```

Comment (tt-dma) [reply tt_metal/hw/inc/wormhole/eth_l1_address_map.h:33]: Yeah might be nice to add a comment at the definition explicitly saying this is why it needs to be like this

**Comment by aliuTT** [comment]
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8349035510

**Pull request merged**